index.html

```diff
@@ -1,390 +1,1471 @@
+        `"serial"`, [=reject=] |promise| with a "{{SecurityError}}"
+        {{DOMException}} and return |promise|.
+      <li>
+        Run the following steps [=in parallel=]:
+        <ol>
+          <li>
+            Let |availablePorts| be the sequence of available serial ports on
+            the system which the user has allowed the site to access as the
+            result of a previous call to {{Serial/requestPort()}}.
+          <li>
+            Let |ports| be the sequence of the {{SerialPort}}s representing the
+            ports in |availablePorts|.
+          <li>
+            [=Queue a global task=] on the [=relevant global object=] of
+            [=this=] using the [=serial port task source=] to [=resolve=]
+            |promise| with |ports|.
+        </ol>
+      <li>Return |promise|.
+    </ol>
+  </section>
+
+  <section>
+    <h3><dfn>onconnect</dfn> attribute</h3>
+
+    {{Serial/onconnect}} is an [=event handler IDL attribute=] for the
+    <dfn>connect</dfn> event type.
+  </section>
+
+  <section>
+    <h3><dfn>ondisconnect</dfn> attribute</h3>
+
+    {{Serial/ondisconnect}} is an [=event handler IDL attribute=] for the
+    <dfn>disconnect</dfn> event type.
+  </section>
 </section>
 
-<section data-dfn-for="SerialPortInfo">
-  ## <dfn>SerialPortInfo</dfn> interface
+<section data-dfn-for="SerialPort">
+  <h2><dfn>SerialPort</dfn> interface</h2>
 
-  ```webidl
+  <pre class="idl">
     [Exposed=(DedicatedWorker,Window), SecureContext]
-    interface SerialPortInfo {
-      maplike&lt;DOMString, DOMString?>;
-    };
-  ```
+    interface SerialPort : EventTarget {
+      attribute EventHandler onconnect;
+      attribute EventHandler ondisconnect;
+      readonly attribute ReadableStream readable;
+      readonly attribute WritableStream writable;
 
-  ### Serial port metadata
+      SerialPortInfo getInfo();
 
-  The value of certain attributes is dependent on the method that the system used
-  to establish the serial port (e.g., USB), and so might not always be available.
+      Promise&lt;undefined> open(SerialOptions options);
+      Promise&lt;undefined> setSignals(optional SerialOutputSignals signals = {});
+      Promise&lt;SerialInputSignals> getSignals();
+      Promise&lt;undefined> close();
+    };
+  </pre>
 
-  ### Recommended serial port metadata
+  Methods on this interface typically complete asynchronously, queuing work on
+  the <dfn>serial port task source</dfn>.
 
-  When available, it is RECOMMENDED that user agents populate a {{SerialPortInfo}}
-  object with the following information. If the value of piece of metadata is
-  unavailable(e.g., if the manufacturer is unknown), then it is RECOMMENDED that
-  that metadata be excluded from the {{SerialPortInfo}}. For interoperability,
-  user agents SHOULD use the values provided in they "Key" column of the
-  <a>table of recommended metadata</a>.
+  <p>
+  The [=get the parent=] algorithm for {{SerialPort}} returns {{Navigator/serial}}.
```

Comment:
This is a bit less precise than it could be. I'd suggest

> returns the same `Serial` instance that is returned by the `SerialPort`'s relevant global object's `Navigator` object's `serial` getter.

Alternately, you could make every `Window` and `WorkerGlobalScope` have a (`<dfn>`ed) associated `Serial` instance, and then have `serial`'s getter steps return that, and the get the parent algorithm return that.